packages/version/src/lib/update-lockfile-version.ts

```diff
@@ -1,32 +1,97 @@
+import { PackageLock, LockDependency } from '@npm/types';
+import fs from 'fs';
+import yaml from 'js-yaml';
 import path from 'path';
+import preferredPM from 'preferred-pm';
 import loadJsonFile from 'load-json-file';
 import writeJsonFile from 'write-json-file';
 import { Package } from '@lerna-lite/core';
+import { Lockfile as PnpmLockfile } from '@pnpm/lockfile-types';
 
-/**
- * From a folder path provided, try to load a `package-lock.json` file if it exists.
- * @param {String} lockFileFolderPath
- * @returns Promise<{path: string; json: Object; lockFileVersion: number; }>
- */
-export async function loadPackageLockFileWhenExists<T = any>(lockFileFolderPath: string) {
+export type NpmLockfile = PackageLock & { packages: { [moduleName: string]: LockDependency } };
+
+export type NpmLockfileInformation = {
+  json: NpmLockfile;
+  path: string;
+  version: number;
+  packageManager: 'npm';
+};
+
+export type PnpmLockfileInformation = {
+  json: PnpmLockfile;
+  path: string;
+  version: number;
+  packageManager: 'pnpm';
+};
+
+export type LockfileInformation = NpmLockfileInformation | PnpmLockfileInformation;
+
+export const isPnpmLockfile = (x: LockfileInformation): x is PnpmLockfileInformation =>
+  x.packageManager === 'pnpm';
+export const isNpmLockfile = (x: LockfileInformation): x is NpmLockfileInformation =>
+  x.packageManager === 'npm';
+
+async function loadYamlFile<T>(filePath: string) {
+  try {
+    const file = await fs.promises.readFile(filePath);
+    return (await yaml.load(`${file}`)) as T;
+  } catch (e) {
+    return undefined;
+  }
+}
+
+async function writeYamlFile(filePath: string, data: unknown) {
   try {
-    const lockFilePath = path.join(lockFileFolderPath, 'package-lock.json');
-    const pkgLockFileObj = await loadJsonFile<T>(lockFilePath);
-    const lockfileVersion = +(pkgLockFileObj?.['lockfileVersion'] ?? 1);
-
-    return {
-      path: lockFilePath,
-      json: pkgLockFileObj,
-      lockfileVersion
-    };
-  } catch (error) { } // eslint-disable-line
+    const str = yaml.dump(data);
+    await fs.promises.writeFile(filePath, str);
+    // eslint-disable-next-line no-empty
+  } catch (e) {}
+}
+
+async function loadNpmLockfile(cwd: string): Promise<LockfileInformation> {
+  const lockfilePath = path.join(cwd, 'package-lock.json');
+  const json = await loadJsonFile<NpmLockfile>(lockfilePath);
+  const version = +(json?.['lockfileVersion'] ?? 1);
+
+  return {
+    json,
+    version,
+    path: lockfilePath,
+    packageManager: 'npm',
+  };
+}
+
+async function loadPnpmLockfile(cwd: string): Promise<LockfileInformation> {
+  const lockfilePath = path.join(cwd, 'pnpm-lock.yaml');
+  const json = (await loadYamlFile(lockfilePath)) as PnpmLockfile;
+  const version = +(json?.['lockfileVersion'] ?? 1);
+
+  return {
+    json,
+    version,
+    path: lockfilePath,
+    packageManager: 'pnpm',
+  };
+}
+
+export async function loadLockfile(cwd: string) {
+  const pm = await preferredPM(cwd);
```

Comment:
is that new dependency really necessary? We already have the `npmClient` command option for this purpose and I would assume that to be sufficient. So I'm not too keen on adding a new lib (of 5kb) when `npmClient` is probably enough, or the original code to just trying loading the file which is what you had in your original PR.

Reply:
What that package does is checking which package manager you are using, I cannot find any `npmClient` that you are talking about?

Reply:
Yea sure, I can rebuild that!

Reply:
hey I looked at the code of `preferred-pm` and it's not 5kb (not sure why it shows 5kb on npm), however it uses [which-pm](https://github.com/zkochan/packages/tree/master/which-pm) created by the same person and I would suggest you to use that one instead, it's quite small and it has less dependencies compare to `preferred-pm` and I assuming it would do what you want to achieve just the same and I agree that it would be easier to use `which-pm` compare to us trying to read every possible lock file.